File: CPE/CPE49/23-UVA11063.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 105;
const int M = 2e4 + 5;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    bool flag;
    int n, kase = 1, arr[N], cnt[M];
    while (cin >> n) {
        for (int i = 0; i < n; i++) cin >> arr[i];
        flag = arr[0] >= 1;
        for (int i = 0; i < n-1; i++) {
            if (arr[i] >= arr[i+1]) {
                flag = false;
                break;
            }
        }
        memset(cnt, 0, sizeof(cnt));
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                if (cnt[arr[i] + arr[j]] > 0) {
                    flag = false;
                    break;
                }
                cnt[arr[i] + arr[j]]++;
            }
        }
        cout << "Case #" << kase++ << ": It is " << (flag ? "a B2-Sequence." : "not a B2-Sequence.") << endl << endl;
    }
    return 0;
}

File: CPE/CPE49/23-UVA11063.py
from collections import Counter

tc = 1
while True:
    try:
        line = input()
        while not line:
            line = input()
        n = int(line)
        arr = list(map(int, input().split()))
    except EOFError:
        break
    flag = True if arr[0] >= 1 else False
    for i in range(n-1):
        if arr[i] >= arr[i+1]:
            flag = False
            break

    cnt = Counter()
    for i in range(n):
        for j in range(i, n):
            if cnt[arr[i] + arr[j]] > 0:
                flag = False
                break
            cnt[arr[i] + arr[j]] += 1
    print(f"Case #{tc}: It is ", end="")
    print("not a B2-Sequence." if not flag else "a B2-Sequence.", end="\n\n")
    tc += 1